drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +515,313 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
+				return false, err
+			}
+			if !isBindMounted {
+				return false, fmt.Errorf("volume [%s] is not bind mounted", vol.ID)
+			}
+			continue // skip retrying
+		} else {
+			_, err = task.DoRetryWithTimeout(t, defaultVmMountCheckTimeout, defaultVmMountCheckRetryInterval)
+			if err != nil {
+				return false, err
+			}
+		}
+	}
+	log.Infof("Successfully verified bind mount for VM [%s] in namespace [%s]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace)
+	return true, nil
+}
+
+// GetNodeOfVM returns nodename on which VM is running
+func GetNodeOfVM(virtualMachineCtx kubevirtv1.VirtualMachine) (string, error) {
+	vmi, err := kubevirt.Instance().GetVirtualMachineInstance(context1.TODO(), virtualMachineCtx.Name, virtualMachineCtx.Namespace)
+	if err != nil {
+		return "", err
+	}
+	log.InfoD("NodeName: %s", vmi.Status.NodeName)
+	return vmi.Status.NodeName, nil
+}
+
+// isNodeInReplicaSet verifies if the replicaset nodes are present in global replicaset
+func IsNodeInReplicaSet(replicaset []*api.ReplicaSet, globalReplicSet []*api.ReplicaSet) (bool, error) {
+	for _, rs := range replicaset {
+		found := false
+	outerLoop:
+		for _, rsNode := range rs.Nodes {
+			for _, grs := range globalReplicSet {
+				if len(grs.Nodes) != len(rs.Nodes) {
+					return false, fmt.Errorf("replicaset mismatch node count mismatch")
+				}
+				for _, grsNode := range grs.Nodes {
+					if grsNode == rsNode {
+						found = true
+						break outerLoop
+					}
+				}
+			}
+		}
+		if !found {
+			return false, fmt.Errorf("replicaset mismatch node not found in global replicaset")
+		}
+	}
+	return true, nil
+
+}
+
+// Get mount type (nfs or bind) of the VM disk
+func getVMDiskMountType(pod *corev1.Pod, vmDisk *volume.Volume, volumeDriver volume.Driver, diskName string) (string, error) {
+	podNamespacedName := pod.Namespace + "/" + pod.Name
+	log.Infof("Checking the mount type of %s in pod %s", vmDisk, podNamespacedName)
+
+	// Sample output if the volume is bind-mounted: (vmDisk.diskName is "rootdisk" in this example)
+	// $ kubectl exec -it virt-launcher-fedora-communist-toucan-jfw7n -- mount
+	// ...
+	// /dev/pxd/pxd365793461222635857 on /run/kubevirt-private/vmi-disks/rootdisk type ext4 (rw,relatime,seclabel,discard)
+	// ...
+	volInspect, err := volumeDriver.InspectVolume(vmDisk.ID)
+
+	bindMountRE := regexp.MustCompile(fmt.Sprintf("/dev/pxd/pxd%s on .*%s type (ext4|xfs)",
+		volInspect.Id, diskName))
+
+	// Sample output if the volume is nfs-mounted: (vmDisk.diskName is "rootdisk" in this example)
+	// $ kubectl exec -it virt-launcher-fedora-communist-toucan-bqcrp -- mount
+	// ...
+	// 172.30.194.11:/var/lib/osd/pxns/365793461222635857 on /run/kubevirt-private/vmi-disks/rootdisk type nfs (...)
+	// ...
+	nfsMountRE := regexp.MustCompile(fmt.Sprintf(":/var/lib/osd/pxns/%s on .*%s type nfs",
+		volInspect.Id, diskName))
+
+	cmd := []string{"mount"}
+	output, err := core.Instance().RunCommandInPod(cmd, pod.Name, "compute", pod.Namespace)
```

Comment:
why "compute"? Does "" not work to use the default container?

Reply:
Changed it to "" will check if this works.